modules/library/xml/src/main/java/org/geotools/xml/resolver/SchemaCache.java

```diff
@@ -464,18 +460,16 @@ static boolean isSuitableDirectoryToContainCache(File directory) {
         if ((new File(directory, CACHE_DIRECTORY_NAME)).isDirectory()) {
             return true;
         }
-        for (String filename : GEOSERVER_DATA_DIRECTORY_FILENAMES) {
-            File file = new File(directory, filename);
-            if (!file.isFile()) {
-                return false;
-            }
-        }
         for (String subdirectory : GEOSERVER_DATA_DIRECTORY_SUBDIRECTORIES) {
             File dir = new File(directory, subdirectory);
             if (!dir.isDirectory()) {
                 return false;
             }
         }
+        // so there is a workspaces directory, lets check default.xml file
+        File workspacesDir = new File(directory, "workspaces");
+        File defaultXmlFile = new File(workspacesDir, "default.xml");
+        if (!defaultXmlFile.exists()) return false;
```

Comment:
A bit confused, if the data dir is empty, truly empty, there will be no workspace either.... but then you cannot configure a store. Maybe it's a "not so empty" case where at least a workspace has been created?

Reply:
Since former dir structure check was based on some files present and no file is present on clean data directory, the algorithm was simplified to check the required workspaces folder only (since app-schema store requires a workspace at least).  Indeed the issue should say that the data dir is files-empty, but no so empty.